src/precice/config/ParticipantConfiguration.cpp

```diff
@@ -194,65 +194,66 @@ ParticipantConfiguration::ParticipantConfiguration(
-    doc += "By default, the participant's MPI_COM_WORLD is reused.";
-    doc += "Use this tag to use TCP/IP sockets instead.";
-    tagMaster.setDocumentation(doc);
-
-    auto attrPort = makeXMLAttribute("port", 0)
-                        .setDocumentation(
-                            "Port number (16-bit unsigned integer) to be used for socket "
-                            "communication. The default is \"0\", what means that OS will "
-                            "dynamically search for a free port (if at least one exists) and "
-                            "bind it automatically.");
-    tagMaster.addAttribute(attrPort);
-
-    auto attrNetwork = makeXMLAttribute(ATTR_NETWORK, utils::networking::loopbackInterfaceName())
-                           .setDocumentation(
-                               "Interface name to be used for socket communication. "
-                               "Default is the canonical name of the loopback interface of your platform. "
-                               "Might be different on supercomputing systems, e.g. \"ib0\" "
-                               "for the InfiniBand on SuperMUC. ");
-    tagMaster.addAttribute(attrNetwork);
-
-    auto attrExchangeDirectory = makeXMLAttribute(ATTR_EXCHANGE_DIRECTORY, "")
-                                     .setDocumentation(
-                                         "Directory where connection information is exchanged. By default, the "
-                                         "directory of startup is chosen.");
-    tagMaster.addAttribute(attrExchangeDirectory);
-
-    intraCommTags.push_back(tagMaster);
-  }
-  {
-    XMLTag tagMaster(*this, "mpi", intraCommOcc, TAG_MASTER);
-    doc = "A solver in parallel needs a communication between its ranks. ";
-    doc += "By default, the participant's MPI_COM_WORLD is reused.";
-    doc += "Use this tag to use MPI with separated communication spaces instead instead.";
-    tagMaster.setDocumentation(doc);
-
-    auto attrExchangeDirectory = makeXMLAttribute(ATTR_EXCHANGE_DIRECTORY, "")
-                                     .setDocumentation(
-                                         "Directory where connection information is exchanged. By default, the "
-                                         "directory of startup is chosen.");
-    tagMaster.addAttribute(attrExchangeDirectory);
-
-    intraCommTags.push_back(tagMaster);
-  }
-  {
-    XMLTag tagMaster(*this, "mpi-single", intraCommOcc, TAG_MASTER);
-    doc = "A solver in parallel needs a communication between its ranks. ";
-    doc += "By default (which is this option), the participant's MPI_COM_WORLD is reused.";
-    doc += "This tag is only used to ensure backwards compatibility.";
-    tagMaster.setDocumentation(doc);
-
-    intraCommTags.push_back(tagMaster);
-  }
-  for (XMLTag &tagMaster : intraCommTags) {
-    tag.addSubtag(tagMaster);
+  for (std::string tag_name : {TAG_MASTER, TAG_PRIMARY_RANK}) {
```

Comment:
Just a reminder: please open an issue such that we remove the `TAG_MASTER` here with v3.0.